subsys/disk/disk_access_spi_sdhc.c

```diff
@@ -748,6 +748,88 @@ static int sdhc_spi_write(struct sdhc_spi_data *data,
 	return err;
 }
 
+/* this function is optimized to write multiple blocks */
+static int sdhc_spi_write_multi(struct sdhc_spi_data *data,
+	const uint8_t *buf, uint32_t sector, uint32_t count)
+{
+	int err;
+	uint32_t addr;
+	uint8_t block[SDHC_CRC16_SIZE];
+
+	err = sdhc_map_disk_status(data->status);
+	if (err != 0) {
+		return err;
+	}
+
+	spi_release(data->spi, &data->cfg);
+
```

Comment:
I don't believe you need to release the SPI controller here?

Reply:
@FRASTM ^ ?

Reply:
line removed

Reply:
And reintroduced?